internal/core/data/v2/controller/http/event_test.go

```diff
@@ -0,0 +1,187 @@
+	noReadingOrigin.Event.Readings[0].Origin = 0
+	noReadingValueType := validRequest
+	noReadingValueType.Event.Readings = []dtos.BaseReading{testReading}
+	noReadingValueType.Event.Readings[0].ValueType = ""
+	noSimpleValue := validRequest
+	noSimpleValue.Event.Readings = []dtos.BaseReading{testReading}
+	noSimpleValue.Event.Readings[0].Value = ""
+	noSimpleFloatEnconding := validRequest
+	noSimpleFloatEnconding.Event.Readings = []dtos.BaseReading{testReading}
+	noSimpleFloatEnconding.Event.Readings[0].ValueType = dtos.ValueTypeFloat32
+	noSimpleFloatEnconding.Event.Readings[0].FloatEncoding = ""
+	noBinaryValue := validRequest
+	noBinaryValue.Event.Readings = []dtos.BaseReading{{
+		DeviceName: TestDeviceName,
+		Name:       TestDeviceResourceName,
+		Origin:     TestOriginTime,
+		ValueType:  dtos.ValueTypeBinary,
+		BinaryReading: dtos.BinaryReading{
+			BinaryValue: []byte{},
+			MediaType:   TestBinaryReadingMediaType,
+		},
+	}}
+	noBinaryMediaType := validRequest
+	noBinaryMediaType.Event.Readings = []dtos.BaseReading{{
+		DeviceName: TestDeviceName,
+		Name:       TestDeviceResourceName,
+		Origin:     TestOriginTime,
+		ValueType:  dtos.ValueTypeBinary,
+		BinaryReading: dtos.BinaryReading{
+			BinaryValue: []byte(TestReadingBinaryValue),
+			MediaType:   "",
+		},
+	}}
+
+	tests := []struct {
+		Name               string
+		Request            []requests.AddEventRequest
+		ErrorExpected      bool
+		ExpectedStatusCode int
+	}{
+		{"Valid - AddEventRequest", []requests.AddEventRequest{validRequest}, false, http.StatusCreated},
+		{"Invalid - No RequestId", []requests.AddEventRequest{noRequestId}, true, http.StatusBadRequest},
+		{"Invalid - No Event", []requests.AddEventRequest{noEvent}, true, http.StatusBadRequest},
+		{"Invalid - No Event DeviceName", []requests.AddEventRequest{noEventDevice}, true, http.StatusBadRequest},
+		{"Invalid - No Event Origin", []requests.AddEventRequest{noEventOrigin}, true, http.StatusBadRequest},
+		{"Invalid - No Reading", []requests.AddEventRequest{noReading}, true, http.StatusBadRequest},
+		{"Invalid - No Reading DeviceName", []requests.AddEventRequest{noReadingDevice}, true, http.StatusBadRequest},
+		{"Invalid - No Reading Name", []requests.AddEventRequest{noReadingName}, true, http.StatusBadRequest},
+		{"Invalid - No Reading Origin", []requests.AddEventRequest{noReadingOrigin}, true, http.StatusBadRequest},
+		{"Invalid - No Reading ValueType", []requests.AddEventRequest{noReadingValueType}, true, http.StatusBadRequest},
+		{"Invalid - No SimpleReading Value", []requests.AddEventRequest{noSimpleValue}, true, http.StatusBadRequest},
+		{"Invalid - No SimpleReading FloatEncoding", []requests.AddEventRequest{noSimpleFloatEnconding}, true, http.StatusBadRequest},
+		{"Invalid - No BinaryReading BinaryValue", []requests.AddEventRequest{noBinaryValue}, true, http.StatusBadRequest},
+		{"Invalid - No BinaryReading MediaType", []requests.AddEventRequest{noBinaryMediaType}, true, http.StatusBadRequest},
```

Comment:
Add a test case for invalid `valueType` , passing something like `"ValueTypeBad"` as the `valueType`

Reply:
Updated to use the latest `go-mod-core-contracts` version(v0.1.72) and added the invalid `valueType` testcase as suggested.